cpp/include/cudf/detail/hashing.hpp

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/hashing.hpp>
+
+namespace cudf {
+namespace detail {
+
+/** --------------------------------------------------------------------------*
+ * @brief Partitions a table into the desired number of partitions according to
+ * the hash value of each row. 
+ *
+ * Computes the hash values of the rows in the specified columns of the input
+ * columns and bins the hash values into the desired number of partitions.
+ * Rearranges the input columns such that rows with hash values in the same bin
```

Comment:
Does it sort them? Or just group by hash value? Maybe this could be more explicit?

```suggestion
 * Sorts the contents of the output columns so that rows with hash values in the same bin
```

Reply:
It doesn't actually sort them. How does this sound?
> Partitions rows of `input` into `num_partitions` bins based on the hash
> value of the columns specified by `columns_to_hash`. Rows partitioned into
> the same bin are grouped together into a new table. Returns a vector
> containing `num_partitions` new tables.